conbench/tests/entities/test_commit.py

```diff
@@ -135,9 +159,16 @@ def test_get_github_commit_and_fork_point_sha_pull_request(branch, pr_number):
         "branch": "dianaclarke:ARROW-13266",
         "fork_point_sha": "780e95c512d63bbea1e040af0eb44a0bf63c4d72",
     }
-    assert (
-        get_github_commit(repo, branch=branch, sha=sha, pr_number=pr_number) == expected
-    )
+    try:
+        result = get_github_commit(repo, branch=branch, sha=sha, pr_number=pr_number)
+    except Exception as exc:
+        log.info("exc during get_github_commit(): %s", exc)
+        if "403 Client Error" in str(exc):
+            pytest.skip("GitHub API rate limit seen, skip test")
```

Comment:
Of course this bakes in the assumption that `"403 Client Error"` corresponds to the API rate limit error(s).

I am rather certain that the `exc` object holds a reference to the HTTP error response object in case `exc` derives from `request.exceptions.HTTPError`. That is, we could _also_ inspect the error response body here and do string-matching on `API rate limit`, if we ever feel like that's necessary.